### PR TITLE
fixes #559 ( Fototoon don't handle window resize )

### LINKED ISSUE #559: Fototoon don't handle window resize
Fototoon activity don't handle window resize event. So, if the activity is open with a small window then resized, the board keep the same size.

ft

### PR DESCRIPTION
**Pull request by anshuldutt21**
fixes #559

### CODE REVIEW

**Comment by s-mag** [comment]
@anshuldutt21 It does not work as expected. The size should decrease on decreasing width and title box should also resize in the same proportion. See [here.](https://gfycat.com/coldcoarsearcticfox)

File: activities/Fototoon.activity/js/activity.js [hunk 1]
```diff
@@ -39,6 +39,8 @@ define(["sugar-web/activity/activity","sugar-web/datastore","sugar-web/env","tex
         // remove 5 more to be sure no scrollbars are visible
         mainCanvas.height = window.innerHeight - sugarCellSize - 5;
         mainCanvas.width = mainCanvas.height * 4 / 3;
+
+        console.log(mainCanvas.width);
```
Comment (llaske) [inline activities/Fototoon.activity/js/activity.js]: Dot not let a log message

File: activities/Fototoon.activity/js/activity.js [hunk 2]
```diff
@@ -66,7 +68,6 @@ define(["sugar-web/activity/activity","sugar-web/datastore","sugar-web/env","tex
         toonModel.init();
         toonModel.attachPageCounterViewer(pageCounter);
         toonModel.attachPrevNextButtons(previousButton, nextButton);
-
```
Comment (llaske) [inline activities/Fototoon.activity/js/activity.js]: No need to change that.

File: activities/Fototoon.activity/js/toon.js [hunk 3]
```diff
@@ -822,7 +822,7 @@ define(["easel","sugar-web/datastore","sugar-web/env","webL10n","humane"], funct
             this._textView.lineWidth = this._globe._width * 2;
             this._textView.x = this._globe._x;
             this._textView.y = this._globe._y -
-                this._textView.getMeasuredHeight() / 2;
```
Comment (llaske) [inline activities/Fototoon.activity/js/toon.js]: You don't need to change this file.

**Comment by anshuldutt21** [comment]
Hi @llaske can you please review it once again. I have changed the proportion of title with respect to the canvas. The canvas size changes on resizing but the title size doesnt unless refreshed even when I call the init method. I think we can add the canvas resize feature only.

**Comment by s-mag** [comment]
@anshuldutt Please remove log messages

**Comment by s-mag** [comment]
@anshuldutt21 you should not use init in resize function. This resets the canvas or the title. See here:-

ezgif com-video-to-gif (11)

**Comment by anshuldutt21** [comment]
@s-mag I cant remove the init function or else a blank canvas is shown. If I change the width and height of canvas I would have to pass it to the new Toon model. Can you suggest any method I could use to remove resetting.

**Comment by s-mag** [comment]
@anshuldutt21 I am not sure but using media queries and scaling the canvas might help...

**Comment by s-mag** [comment]
@anshuldutt21 Nice but it does not solve the issue completely... The title box should resize.

**Comment by anshuldutt21** [comment]
@s-mag I have changed the width and height of the title box according to the canvas, but it is still not resizing. I think there is some other issue in it which works on reloading but not on resizing.

**Comment by anshuldutt21** [comment]
@llaske Can you please review the PR

**Comment by llaske** [comment]
It's a good start but I think it will be nice if when the window is resized the proportion (relative position) is keep.

image

**Comment by anshuldutt21** [comment]
@llaske if I use the init method for resizing the title box , all the initial data corresponding to it will be lost. So I think the best way for now is to leave it like that because otherwise it will keep on reloading and all the data will be lost. Also, We could create another issue for the title resizing thing.

**Comment by llaske** [comment]
@anshuldutt21  I'm not agree. I don't understand why resize necessarily need that all data will be lost. Currently, images are resize to fit the screen and no data are lost.

**Comment by llaske** [comment]
Close due to lack of activity. @anshuldutt21 feel free to reopen it if need.

**Pull request closed without merging**